embrace-android-features/src/main/kotlin/io/embrace/android/embracesdk/internal/injection/NativeFeatureModuleImpl.kt

```diff
@@ -95,6 +103,52 @@ internal class NativeFeatureModuleImpl(
         }
     }
 
+    override val nativeCrashHandlerInstaller: NativeCrashHandlerInstaller? by singleton {
+        if (configModule.configService.autoDataCaptureBehavior.isNativeCrashCaptureEnabled()) {
+            NativeCrashHandlerInstallerImpl(
+                configService = configModule.configService,
+                sharedObjectLoader = nativeCoreModule.sharedObjectLoader,
+                logger = initModule.logger,
+                repository = embraceNdkServiceRepository,
+                delegate = JniDelegateImpl(),
+                backgroundWorker = workerThreadModule.backgroundWorker(Worker.Background.IoRegWorker),
+                nativeInstallMessageProvider = {
```

Comment:
This is my bad as I originally suggested a `Provider`, but if everything we need to create the `nativeInstallMessage` is within the module, then we can probably just pass in the created object directly?

Reply:
Yeah, that'd probably be better. I don't think we need to wait for initialization to access sessionId, appState and is32BitDevice.